src/axom/core/ArrayBase.hpp

```diff
@@ -168,13 +168,28 @@ class ArrayBase
    *
    * \param [in] shape Array size in each direction.
    */
-  AXOM_HOST_DEVICE ArrayBase(const StackArray<IndexType, DIM>& shape)
+  AXOM_HOST_DEVICE ArrayBase(const StackArray<IndexType, DIM>& shape,
+                             int spacing = 1)
```

Comment:
Please add a comment about the 'spacing' arg.

Reply:
@rhornung67 and @publixsubfan  Spacing is the stride in the fastest changing direction.  It's usually 1, but we introduced spacing to let it be different.  Now that we allow custom strides in all directions, it would make sense to remove the terminology.

Reply:
I think min_stride is clearer than spacing. Also, the term 'space' is vague in this context and is also an overloaded term since we use it to refer to memory space.